### PR TITLE
Add resources, sources, and references during create component flow

### PR DESCRIPTION
**Pull request by jmickey**
Enable users to add resources, sources, and references within the Create Component UI flow.

- [x] Add resources (file, directory, ociImage)
- [x] Add sources
- [x] Add references

Changes:

- Added buttons for "Add Resource", "Add Source", and "Add Reference"
- Adding a resources allows the user to specify a resource type, current supported types are "file", "directory", and "ociImage".
- Adding a source currently support a gitHub source.
- Adding a reference supports a basic reference specification of `name`, `version`, and `componentName`. Support for specifying a digest can be completed later.
- If an error occurs adding any of the above the process exits early and the user will need to rectify.

Current Limitations:

- Once you add a resource, source, or reference, you can't remove them. This requires some additional considerations.
- Only basic input validation is completed, which only validates that the fields are not empty.

### CODE REVIEW

File: src/commands/addResources.ts
Comment (phoban01) [inline src/commands/addResources.ts:1]: I think using the a resource file would be a more straightforward way to approach this, We could just serialize the resource, create a temporary file and pass it to the CLI

File: src/commands/addSource.ts
Comment (phoban01) [inline src/commands/addSource.ts:1]: Ditto previous comment re input file

File: src/webviews/createComponent.ts [hunk 1]
```diff
@@ -21,16 +22,67 @@ interface CreateComponentWebviewContent {
...
-    componentName: string;
-    version: string;
-    provider: string;
-    scheme: string;
+    fileUri: string;
+    forResourceId: number;
   };
 }
 
+export interface FileResource {
```
Comment (phoban01) [inline src/webviews/createComponent.ts:33]: We should use the ocm types already defined here: https://github.com/open-component-model/vscode-ocm-tools/blob/main/src/ocm/ocmv3.d.ts

**Review by phoban01 — CHANGES_REQUESTED** [review]
Approach looks good. I think it can be improved by using the input file and leveraging the generated OCM types.

**Comment by phoban01** [comment]
It would be nice to be able to specify a path within the workspace in which to create the component.

**Comment by phoban01** [comment]
For some reason when a component is created using the extension it is not possible to remove it from the treeview, even if it remove the `component-archive directory.

I have validated this using the following process:
- create a component using the CLI 
- refresh "Local Components" treeview
- component is listed in treeview
- delete the component archive directory
- refresh "Local Components" treeview
- component  is removed from treeview
- create component using the extension
- refresh "Local Components" treeview
- component is listed in treeview
- delete the component archive directory
- refresh "Local Components" treeview
- component remains listed in treeview (when it should have been removed)

**Comment by jmickey** [comment]
@phoban01 

> Adding a Source currently doesn't seem to work.

This is resolved.

> For some reason when a component is created using the extension it is not possible to remove it from the treeview

I noticed this too, and tracking down why was pretty difficult, but I believe I have fixed the issue in the latest commit. Refresh behaviour is a little opaque, but I found this nice example from the MS team: https://github.com/microsoft/vscode-extension-samples/blob/main/tree-view-sample/src/nodeDependencies.ts, which helped me resolve this. 

Essentially, the tree wasn't correctly being rebuilt when refresh was called, which in the example is being done in the `getChildren` method. This lead me to notice that the internal `treeViews` state isn't really required, so I removed it. From my manual testing the refresh functionality is now working as intended with the current changes.

> It would be nice to be able to specify a path within the workspace in which to create the component.

Should be easy enough.

**Comment by phoban01** [comment]
Verified that Sources can be added and that refreshing works as expected now 🥇

File: src/commands/createComponent.ts [hunk 2]
```diff
@@ -32,21 +42,73 @@ export async function createComponent(
...
+      if (!res.success) {
+        // return early if a source could not be added.
+        workspaceTreeViewProvider.refresh();
+        return {
+          success: false,
+          message: `Component creation failed adding source: ${res.message}`,
+        };
+      }
+    }
+  }
```
Comment (phoban01) [inline src/commands/createComponent.ts:104]: this is nice 👍🏻

**Review by phoban01 — APPROVED** [review]
Looks great, thanks Josh!

**Pull request merged**